syntax = "proto3";

option java_multiple_files = true;
option java_package = "cn.abellee.cniface.grpc";
option java_outer_classname = "CNIFaceProto";
option objc_class_prefix = "CNIFACE";

package cniface;


service DetectService {
    rpc detect (DetectRequest) returns ( DetectResponse) {}
}

message DetectRequest {
    string model = 1;
    string faceImageBase64 = 2;
    float score = 3;
}

message DetectResult {
    float score = 1;
    float x = 2;
    float y = 3;
    float w = 4;
    float h = 5;
    repeated float kps = 6;
    float maskScore = 7;
}

message DetectResponse {
    int32 code = 1;
    string message = 2;
    repeated DetectResult results = 3;
}


service RecognitionService {
    rpc extractFeature (ExtractFeatureRequest) returns ( ExtractFeatureResponse) {}
    rpc similarity (SimilarityRequest) returns ( SimilarityResponse) {}
}

message ExtractFeatureRequest {
    string model = 1;
    string faceImageBase64 = 2;
    repeated float kps = 3;
}

message ExtractFeatureResponse {
    int32 code = 1;
    string message = 2;
    repeated float feature = 3;
}

message SimilarityRequest {
    repeated float feature1 = 1;
    repeated float feature2 = 2;
}

message SimilarityResponse {
    int32 code = 1;
    string message = 2;
    float similarity = 3;
}


service AttributeService {
    rpc genderAge (GenderAgeRequest) returns ( GenderAgeResponse) {}
}

message GenderAgeRequest {
    string model = 1;
    string faceImageBase64 = 2;
    repeated float kps = 3;
}

message GenderAgeResponse {
    int32 code = 1;
    string message = 2;
    int32 gender = 3;
    int32 age = 4;
}

